spec/alephant/broker/cache/cached_object_spec.rb

```diff
@@ -0,0 +1,111 @@
+require 'spec_helper'
+
+RSpec.describe Alephant::Broker::Cache::CachedObject do
+  subject { described_class.new(s3_obj) }
+
+  let(:config)        { {} }
+  let(:last_modified) { Time.parse('Mon, 11 Apr 2016 10:39:57 GMT') }
+  let(:ttl)           { 15 }
+
+  let(:s3_obj) do
+    {
+      content:      'Test',
+      content_type: 'test/content',
+      meta:         {
+        :ttl                  => ttl,
+        :head_ETag            => '123',
+        :'head_Last-Modified' => last_modified.to_s
+      }
+    }
+  end
+
+  before do
+    allow_any_instance_of(Logger).to receive(:info)
+    allow(Alephant::Broker).to receive(:config).and_return(config)
+  end
+
+  describe '#updated' do
+    it 'extracts the #updated time from the S3 object' do
+      Timecop.freeze do
+        expect(subject.updated).to eq(last_modified)
+      end
+    end
+
+    context 'when there is no Last-Modified on the S3 object' do
+      let(:last_modified) { nil }
+
+      it 'sets #updated to now' do
+        now = Time.parse('Mon, 31 May 2016 12:00:00 GMT')
+
+        Timecop.freeze(now) do
+          expect(subject.updated).to eq(now)
+        end
+      end
+    end
+  end
+
+  describe '#ttl' do
+    it 'extracts the #ttl from the S3 object' do
+      expect(subject.ttl).to eq(ttl)
+    end
+
+    context 'when there is no TTL on the S3 object' do
+      let(:ttl) { nil }
+
+      context 'and a default cache TTL has been configured' do
+        let(:config) { { revalidate_cache_ttl: 100 } }
+
+        it 'sets the #ttl to the configured value' do
+          expect(subject.ttl).to eq(100)
+        end
+      end
+
+      context 'and a default cache TTL has NOT been configured' do
+        it 'sets the #ttl to a default (in code) value' do
+          expect(subject.ttl).to eq(described_class::DEFAULT_TTL)
+        end
+      end
+    end
+  end
+
+  describe '#update' do
+    let(:new_content) do
+      {
+        content:      'Test - NEW',
+        content_type: 'test/content',
+        meta:         {
+          :ttl                  => ttl,
+          :head_ETag            => '123',
+          :'head_Last-Modified' => (last_modified + 100).to_s
```

Comment:
This spec has nothing to do with the `ttl` - it's just changing the object so you can see a difference between the new and the old object value...

Will remove some of this so there is no confusion